wolfcrypt/src/sp_int.c

```diff
@@ -12719,6 +12719,10 @@ int sp_tohex(sp_int* a, char* str)
                     break;
                 }
                 else if (j == 0) {
+                    if (i == 0) {
```

Comment:
I didn't think we needed to check for zero here when I made my first pass at fixing this because of the sp_iszero check at line 12700. However, that function only returns true if a->used is 0. In Guido's latest reproducer, a->used is 1 and a->dp[0] is 0, which causes a failure because we aren't checking for zero after sp_iszero. This change fixes that.

Reply:
How did a->used get to be greater than 0?

Reply:
I'm unsure, but I can go back through with gdb and find out.

Reply:
The reproducer is here https://wolfssl.zendesk.com/agent/tickets/11434/. After calling sp_mont_norm, B->used is 1 amd B->dp[0] is 0. I'm unfamiliar with what exactly is going on in the math functions, though, so I don't know if this result is unexpected.

Reply:
Fix sp_mont_norm() by calling sp_clamp() at the end.